### PR TITLE
Auth

### PR DESCRIPTION
**Pull request by NamrataRaikwar2002**
- user can signup
- user can login/logout
- authentication require for private routes.

### CODE REVIEW

File: src/redux/thunk/authThunk.js [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+const login = createAsyncThunk(
+  'auth/login',
+  async ({ username, password }, { rejectWithValue }) => {
+    console.log(password)
+    try {
+      const response = await axios.post('/api/auth/login', {
+        username,
+        password,
+      })
+      console.log(response)
```
Comment (Ayushi20-19) [inline src/redux/thunk/authThunk.js:35]: u can remove consoles,

File: src/redux/thunk/authThunk.js [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+        username,
+        password,
+        firstName,
+        lastName,
+      })
+      const data = { data: response.data, status: response.status }
+      return data
+    } catch (error) {
+      return rejectWithValue({
+        data: error.response.data,
```
Comment (Ayushi20-19) [inline src/redux/thunk/authThunk.js:19]: i think these values can be set in extrareducer  reject call

File: src/RequireAuth.js [hunk 3]
```diff
@@ -0,0 +1,12 @@
+import { useSelector } from 'react-redux'
+import { Navigate, useLocation } from 'react-router'
+export const RequireAuth = ({ children }) => {
+  const location = useLocation()
+  const { token } = useSelector((state) => state.auth)
+  console.log(token)
```
Comment (Ayushi20-19) [inline src/RequireAuth.js:6]: u can remove console

File: src/Pages/Auth/Signup.jsx [hunk 4]
```diff
@@ -1,11 +1,69 @@
...
+      const response = await dispatch(signup(signupUser))
+      setSignupUser({
+        firstName: '',
+        lastName: '',
+        email: '',
+        password: '',
+        confirmPassword: '',
+      })
+      if (response?.payload?.status === 201) {
+        console.log(response, 'signup page')
```
Comment (Ayushi20-19) [inline src/Pages/Auth/Signup.jsx:48]: here also

**Review by Ayushi20-19 — APPROVED** [review]
just a few minor suggestions, else good work,🎉
-  code is good and working as expected,
-  good error handling,
- and you can remove unnecessary consoles.

**Review by Tanjum786 — APPROVED** [review]

**Pull request merged**